lib/monitoring/aws-rds/RdsClusterMetricFactory.ts

```diff
@@ -27,6 +32,14 @@ export class RdsClusterMetricFactory {
     props: RdsClusterMetricFactoryProps
   ) {
     this.metricFactory = metricFactory;
+    this.cluster = props.cluster || undefined;
```

Comment:
This is where I try to set the cluster attribute based on the props, but for some reason it loses its typing of being a `ServerlessCluster`